api/src/api/comment/comment.controllers.ts

```diff
@@ -28,6 +27,9 @@ const CreateComment = async (req: Request, res: Response, next: NextFunction) =>
 const GetComment = async (req: Request, res: Response, next: NextFunction) => {
   try {
     const { id } = req.params;
+    if (!id) {
+      throw new HttpError("Missing required fields", 400);
+    }
```

Comment:
What I meant previously was that those `if (!id)` check are unnecessary because if an user omits the id (e.g. `comments/1` to `comments/` it would be a different route.

What we need to throw 404 for is when there is no matching resource with the provided id: user trying to access `comments/99` but we only have 98 comments in the DB. You should set up a check as follow:
```ts
    const comment = await getComment(id);
    if (!comment) { 
        // throw 404 
    }
```

This applies for all routes that deal with `/:id`